lib/auth/auth_with_roles.go

```diff
@@ -1211,6 +1211,7 @@ func (a *ServerWithRoles) listResourcesWithSort(ctx context.Context, req proto.L
 		if err := servers.SortByCustom(req.SortBy); err != nil {
 			return nil, trace.Wrap(err)
 		}
+		servers = types.DeduplicateAppServers(servers)
```

Comment:
For this (and the other ones): why sort and then dedup, rather than dedup then sort?